File: src/display/AlphaImage.java
package display;

import java.awt.Color;
import java.awt.image.BufferedImage;
import java.io.File;
import java.io.IOException;

import javax.imageio.ImageIO;

import main.Settings;

/**
 * a container for name and an image
 * @author McAJBen <McAJBen@gmail.com>
 * @since 1.6
 */
public class AlphaImage {
	
	/**
	 * the name of the file that is loaded
	 */
	private String name;
	
	/**
	 * the file used to load the image
	 */
	private File file;
	
	/**
	 * creates an instance of {@code AlphaImage}
	 * @param folder the folder that contains the file named n
	 * @param n the name of the specific file
	 */
	public AlphaImage(File folder, String n) {
		name = n;
		file = new File(folder.getAbsolutePath() + File.separator + name);
	}
	
	/**
	 * gets the name of the file
	 * @return a {@code String} for the file name
	 */
	public String getName() {
		return name;
	}
	
	/**
	 * gets the Image
	 * @return a {@code BufferedImage} with the file name
	 */
	public BufferedImage getImage() {
		try {
			return ImageIO.read(file);
		} catch (IllegalArgumentException | IOException e) {
			Settings.showError("Cannot load Image \"" + name, e);
			return null;
		}
	}
	
	/**
	 * gets the background color of the image by using the top left corner pixel
	 * @return the color of the top left corner of the image
	 */
	public Color getBGColor() {
		File f = Settings.fileToThumb(file);
		try {
			return new Color(ImageIO.read(f).getRGB(0, 0));
		} catch (IllegalArgumentException | IOException e) {
			Settings.showError("Cannot load Image RGB \"" + name, e);
		}
		return Color.BLACK;
	}
}

File: src/display/AlphaImage.kt
package display

import main.Mode
import util.Labels
import util.Log
import util.Settings
import java.awt.Color
import java.awt.image.BufferedImage
import java.io.File
import javax.imageio.ImageIO

/**
 * a container for name and an image
 * @param mode used to determine the folder that contains the file
 * @param name the name of the specific file
 * @author McAJBen@gmail.com
 * @since 1.6
 */
class AlphaImage(
    mode: Mode,
    val name: String
) {

    /**
     * the file used to load the image
     */
    private val file = File(Settings.getFolder(mode), name)

    /**
     * the thumbnail file for this image
     */
    private val thumbnail = File(Settings.getDataFolder(mode), name)

    /**
     * the `BufferedImage` from the file
     */
    val image: BufferedImage?
        get() {
            var error: OutOfMemoryError? = null
            for (i in 0..49) {
                try {
                    return ImageIO.read(file)
                } catch (e: OutOfMemoryError) {
                    error = e
                    Thread.sleep(10)
                } catch (e: Exception) {
                    Log.error(String.format(Labels.CANNOT_LOAD_IMAGE, name), e)
                }
            }
            if (error != null) {
                Log.error(String.format(Labels.CANNOT_LOAD_IMAGE_ERROR_MULTIPLE, name), error)
            }
            return null
        }

    /**
     * the background color of the image by using the top left corner pixel
     */
    val bGColor: Color
        get() {
            try {
                return Color(ImageIO.read(thumbnail).getRGB(0, 0))
            } catch (e: Exception) {
                Log.error(String.format(Labels.CANNOT_LOAD_IMAGE_RGB, name), e)
            }
            return Color.BLACK
        }

}